### PR TITLE
update feedstock to version 5.0.1

### LINKED ISSUE #9: add user
### Additional comment:

@conda-forge-admin, please add user @username

### PR DESCRIPTION
**Pull request by suzannejin**
Update feedstock to version 5.0.1. This version will be added as addition to the existing version 
@conda-forge-admin, please rerender

Closes #9.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by suzannejin** [comment]
@conda-forge-admin, please rerender

**Comment by suzannejin** [comment]
@conda-forge-admin, please ping team

**Comment by suzannejin** [comment]
@conda-forge-admin, please add bot automerge

**Comment by suzannejin** [comment]
Hello @dpryan79 , could you please review this PR?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -33,16 +32,28 @@ requirements:
...
+    # - bioconductor-aldex2
+    - r-corpcor
     - r-fastcluster
-    - r-ggplot2
-    - r-igraph
+    - r-knitr
+    # - bioconductor-limma
+    - r-ppcor
+    - r-rmarkdown
+    - r-testthat
```
Comment (dbast) [inline recipe/meta.yaml]: hm... should knitr/rmarkdown/testthat really be here as run dependency? they are listed under "suggests" https://github.com/tpq/propr/blob/master/DESCRIPTION#L34 and I guess not required when using the package.

Comment (suzannejin) [reply recipe/meta.yaml]: Hello @dbast , I just removed those dependencies from the recipe. Now it should be fine :)

File: recipe/meta.yaml [hunk 2]
```diff
@@ -33,16 +32,22 @@ requirements:
...
     - {{ native }}gcc-libs         # [win]
     - r-rcpp
+    # - bioconductor-aldex2
+    - r-corpcor
     - r-fastcluster
-    - r-ggplot2
-    - r-igraph
+    # - bioconductor-limma
+    - r-ppcor
+    - r-vegan
```
Comment (mfansler) [inline recipe/meta.yaml]: AFAICT, only `Rcpp` is actually a run dependency. Unlike CRAN, we don't install Suggests packages during build - these are usually needed for vignette rendering and running examples, which we omit.

I suggest this be simplified to:

```suggestion
  host:
    - r-base
    - r-rcpp
  run:
    - {{ native }}gcc-libs         # [win]
    - r-base
    - r-rcpp
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,20 +1,19 @@
...
+  url: "{{ github }}/archive/refs/tags/{{ version }}.tar.gz"
+  sha256: c0a0cf8b007501919bbe0585e412a62902f61c70cc3dfa2b56546428b16db887
 
 build:
   merge_build_host: True  # [win]
-  number: 3
+  number: 0
   rpaths:
     - lib/R/lib/
     - lib/
```
Comment (mfansler) [inline recipe/meta.yaml:19]: Let's take care of the Windows linking warning while we're touching this.

```suggestion
    - lib/
  missing_dso_whitelist:
    - '*/R.dll'  # [win]
```

**Review by mfansler — CHANGES_REQUESTED** [review]
Thanks for taking time to put this together! ❤️ 

I have a couple suggestions before signing off.

File: recipe/meta.yaml (hunk 2 above)
Comment (suzannejin) [reply recipe/meta.yaml]: I actually need to be able to use r-corpcor and r-ppcor during run time. Is it not possible to include them? Indeed these two packages are needed to run some of the correlation matrices that propr compute.

**Comment by suzannejin** [comment]
hey @mfansler, i just declared 'corpcor' and 'ppcor' as packages in imports. 
And submitted the new meta.yaml with only the packages declared in imports.
I guess it should be fine now :)

**Comment by mfansler** [comment]
Sorry to be a stickler about this, but what is the motivation that makes these now be considered essential functionality of the package? The package was configured to run namespace checks for these backend packages, which indicates that they were considered optional functionality. For example, was there some reason that these other packages couldn't previously be direct imports when the package was on CRAN?

Given you are an upstream maintainer, I'm not going to resist including these beyond requesting a fuller explanation. Ultimately, it is at your discretion to choose to have the Conda installation behavior diverge from `remotes::install_github("tpq/propr")`. However, this is atypical and should be done with care. Personally, I would regard changing dependencies as meriting a minor version bump and something to be communicated in a NEWS entry.

BTW, are there no plans to restore this package to CRAN? It seems it [got kicked off because](https://cran-archive.r-project.org/web/checks/2022/2022-11-10_check_results_propr.html) exactly 1 unit test showed loss of consistency with another package. Can that not be reconciled?

**Comment by suzannejin** [comment]
Hey @mfansler , the functionalities related to these two packages, currently moved to import, were only officially introduced from propr v5.0 and beyond. They were previously declared in suggest, but I do think it is proper to have them in imports. 
As for restoring the package to CRAN, I did suggested it to @tpq.

**Review by mfansler — APPROVED** [review]

**Comment by mfansler** [comment]
> They were previously declared in suggest, but I do think it is proper to have them in imports.

It would be helpful to state the reasons. The upstream package still keeps all references to these packages behind a namespace check which is the design pattern for *optional* functionality.

**Pull request merged**